### PR TITLE
User Gallery/Deck of Cards: Fixed #22219

### PR DESCRIPTION
**Pull request by mjansenDatabay**
See: https://www.ilias.de/mantis/view.php?id=22219

### CODE REVIEW

**Comment by Amstutz** [comment]
Hi mjansen. Can you explain in a few words what you are doing here and why? I consulted the linked ticket, but I am not sure I understand the implications here...

**Comment by mjansenDatabay** [comment]
Hi @Amstutz: Currently there is an issue regarding the height of cards (with a different number of sections) in a deck, only between 768px and 991px screen width.

Please check the members gallery in https://ilias.de/test53/goto.php?target=crs_48772 .

The few lines changed in this PR solve this issue.

File: src/UI/templates/default/Card/card.less [hunk 1]
```diff
@@ -47,7 +47,7 @@
 	background-color: @il-secondary-container-bg;
 }
 
-@media (min-width: @screen-md-min) {
+@media (min-width: @screen-sm-min) {
```
Comment (Amstutz) [inline src/UI/templates/default/Card/card.less:50]: +1

File: src/UI/templates/default/Deck/deck.less [hunk 2]
```diff
@@ -21,3 +21,23 @@
     }
   }
 }
+
+@media (min-width: @screen-sm-min) and (max-width: @screen-sm-max) {
```
Comment (Amstutz) [inline src/UI/templates/default/Deck/deck.less]: why is this whole part needed? Why not changing the above to sm-min?

File: src/UI/templates/default/Deck/deck.less [hunk 3]
```diff
@@ -21,3 +21,23 @@
...
+  .row-eq-height {
+
+    display: -webkit-box;
+    display: -webkit-flex;
+    display: -ms-flexbox;
+    display: flex;
```
Comment (Amstutz) [inline src/UI/templates/default/Deck/deck.less]: those first four lines are could be shared with @screen-md-min in a sm-min section, to enhance readability.

File: src/UI/templates/default/Deck/deck.less [hunk 4]
```diff
@@ -21,3 +21,23 @@
...
+    flex: 3;
```
Comment (Amstutz) [inline src/UI/templates/default/Deck/deck.less]: This is shorthand for flex-grow: 3;. Please change to enhance the readability.

why is this needed? why would any .row-eq-height need a separate flex-grow? If they all get grow 3, this would not have any effect, right, or am I missing something (which is very possible).

File: src/UI/templates/default/Deck/deck.less [hunk 5]
```diff
@@ -21,3 +21,23 @@
...
+    flex-wrap: wrap;
```
Comment (Amstutz) [inline src/UI/templates/default/Deck/deck.less]: Why is this only needed for (min-width: @screen-sm-min) and (max-width: @screen-sm-max) and not just for (min-width: @screen-sm-min)? Others should also wrap, right? Again probably I am missing something.

File: src/UI/templates/default/Deck/deck.less [hunk 6]
```diff
@@ -21,3 +21,23 @@
...
+    
+    > div {
+      display: -webkit-box;
+      display: -webkit-flex;
+      display: -ms-flexbox;
```
Comment (Amstutz) [inline src/UI/templates/default/Deck/deck.less]: Why is this only needed here, an not also in the above @screen-md-min section (which then again could be merged with this section here).

File: src/UI/templates/default/Deck/deck.less [hunk 7]
```diff
@@ -21,3 +21,23 @@
...
+      display: flex;
+      flex-grow: 2;
```
Comment (Amstutz) [inline src/UI/templates/default/Deck/deck.less]: Again, I do not get the flex-grow here.

**Review by Amstutz — COMMENTED** [review]
First of all, obviously: Thank you very much for the fix. I saw the according mantis bug. My bad, I was unclear. I was wondering about several lines in the code of which I do not understand the purpose of. I left according comments/questions now.

I know, I am somewhat obsessed with the meaning of those less lines, sry for that. I sure it's worth it in the long run.

**Comment by mjansenDatabay** [comment]
@Amstutz : Please check my latest commit. Setting **flex-wrap: wrap;** seems to be sufficient beside the min-width change.

**Comment by Amstutz** [comment]
Thanks a lot!
